docs/walletkit/android/chain-abstraction.mdx

```diff
@@ -0,0 +1,233 @@
+```
+
+### Get Fulfilment Details
+
+This method is used to get the details of the fulfillment operation.
+
+```kotlin
+@ChainAbstractionExperimentalApi
+fun getFulfilmentDetails(
+  available: Wallet.Model.FulfilmentSuccess.Available,
+  initTransaction: Wallet.Model.Transaction,
+  onSuccess: (Wallet.Model.FulfilmentDetails) -> Unit,
+  onError: (Wallet.Model.Error) -> Unit
+)
+``` 
+
+### Usage
+
+When your wallet receives an `eth_sendTransaction` request, first check if chain abstraction is needed using the `prepare` method, if it is, you need to sign all the fulfilment transactions and broadcast them in parallel. 
+After that, you need to call the `status` method to check the status of the fulfillment operation. 
+
+If the operation is successful, you need to broadcast the initial transaction and await for the transaction hash and receipt. 
+If the operation is not successful, you need to send the JsonRpcError to the dapp and display the error to the user. 
+
+```kotlin
+override fun onSessionRequest(sessionRequest: Wallet.Model.SessionRequest, verifyContext: Wallet.Model.VerifyContext) {
+  if (sessionRequest.request.method == "eth_sendTransaction") {
+    val initialTransaction = Wallet.Model.Transaction(...)
+    
+    WalletKit.prepare(
+      initialTransaction,
+      onSuccess = { fulfilmentSuccess ->
+        when (fulfilmentSuccess) {
+          is Wallet.Model.FulfilmentSuccess.Available -> {
+            //Sign all the fulfilment transactions
+            //Broadcast the fulfilment transactions in parallel
+            //Await for the transaction hashes and receipts
+            
+            //Call the fulfillmentStatus
+            WalletKit.status(fulfilmentId, checkIn,
+                onSuccess = {
+                    //Successfull filfilment
+                    //Broadcast the inittial transaction
+                    //Await for the tx hash ande receipt
+                    //Send the response to the Dapp
+                },
+                onError = {
+                    //Fulfilment error - wallet should send the JsonRpcError to a dapp for given request and display error to the user
+                }
+            )
+          }
+
+          is Wallet.Model.FulfilmentSuccess.NotRequired -> {
+              //Chain abstraction is not required, handle transaction as usual
+          }
+        }
+      },
+      onError = { fulfilmentError ->
+            // One of the possible errors: NoRoutesAvailable, InsufficientFunds, InsufficientGasFunds - wallet should send the JsonRpcError to a dapp for given request and display error to the user
+      }
+   )
+  } else {
+    // Handle transaction as usual
+  }
+}
+```
+## Types
+
+Following are the types that are used in the chain abstraction methods.
+
+```kotlin
+data class Transaction(
+  var from: String,
+  var to: String,
+  var value: String,
+  var gas: String,
+  var gasPrice: String,
```

Comment:
This field is not supported; can we remove?

Reply:
Yup I'll commit new types